### PR TITLE
fix(voip): parse frame info independently of optional RTP extensions

### PR DESCRIPTION
**Pull request by jlucaso1**
## Summary

Follow-up to #1469. Receive orientation still depended on an exact 12-byte extension layout. Valid frame-info-only, reordered and extended metadata therefore fell back to signaling instead of using RTP rotation.

Add an allocation-free frame-info reader and use it after authentication. Preserve the public header structs, exact outgoing encoding and optional bandwidth/timing fields. No values are fabricated for absent optional extensions. AU aggregation, signaling fallback semantics and camera-facing behavior are unchanged.

The captured WASM dispatcher and frame constructor establish the parser discrepancy. The live Android rear-camera extension layout has not been captured, so this PR does not claim to prove the cause of that visual failure.

## Verification

- Observed the regression fail before the fix on reordered metadata, with Rust returning `None` and WASM returning explicit zero.
- Core VoIP suite with `voip-mlow`: 690 passed, one existing diagnostic test ignored. Includes RTP, authenticated fragmented receive, engine fallback and fuzz tests.
- Explicit captured-WASM incoming and existing orientation suites: five tests passed, no skips. Covers 3,840 packets, 285 parser edge cases and eight passthrough pairs.
- Clippy passed for `wacore --features voip-mlow --all-targets` and `oracle-core --all-targets --release`, both with `-D warnings`.
- `cargo fmt --all -- --check` passed.

The parser oracle checks padding, unknown ID 15, lengths 1 through 16, duplicate format precedence and truncated-element behavior. It never writes intermediate frame orientation and invokes no host imports during parser comparisons. Capture identity, function selectors and execution limits are documented in `agent_docs/voip_media_oracle.md`.

## Limits

The broader `cargo test --release -p oracle-core` run reached the existing threaded-engine suite but exceeded the 10-minute command limit. The targeted orientation suites passed separately. A release-only core run also encountered an existing test that expects a debug assertion to panic; the full normal-profile VoIP run passed. No thread-harness or unrelated test changes are included.

### CODE REVIEW

File: agent_docs/binary_size_ci.md [hunk 1]
```diff
@@ -23,12 +23,43 @@ Do NOT switch any metric to rlib size: rlibs carry un-monomorphized generics plu
...
-- Sizes are only comparable under the same pinned toolchain. A `rust-toolchain.toml` bump legitimately moves every metric — expect a gate hit and use the label.
+- Sizes are only comparable under the same pinned toolchain. A toolchain change requires a base measurement with the matching compiler; `size-increase-ok` cannot override missing or incomparable measurements. The selector skips other-compiler candidates at the same base SHA, not metadata or provenance failures.
 - `cargo bloat` exits 0 even on analysis errors; the measure script validates its JSON instead of trusting the exit code.
 - Fork PRs run with a read-only token: they get the job summary and the gate, but no PR comment.
-- Local run: `cargo xt ci measure-binary-size --out-dir size-out` (add `--skip-build` to reuse an existing release build).
+- CI invokes the dispatcher with `cargo run --locked --quiet -p whatsapp-xtask --`. The build, cargo-bloat and cargo-llvm-lines also use `--locked`. Measurement refuses lockfile changes, and CI checks the tracked worktree before uploading. Fix manifest/lock inconsistencies explicitly rather than letting measurement rewrite them and break the subsequent gh-pages checkout.
```
[the remark this answers is not in the document]
Comment (jlucaso1) [reply agent_docs/binary_size_ci.md:33]: Fixed in 876fa85b. The PR gate documentation now uses cargo run --locked --quiet -p whatsapp-xtask -- ci binary-size-report, consistent with the workflow and local measurement instructions. No size budget or lockfile-change enforcement was relaxed. Tooling tests, Clippy, formatting and actionlint passed.

File: tools/oracle-core/tests/incoming_orientation_probe.rs [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+use wasmtime::Val;
+
+mod common;
+
+#[test]
+fn incoming_rtp_metadata_to_frame_descriptor() -> anyhow::Result<()> {
+    let Some(bytes) = common::capture("JgwtTQVeWPm")? else {
+        eprintln!("skipping: JgwtTQVeWPm unavailable (set WA_WASM_DIR)");
+        return Ok(());
+    };
```
[the remark this answers is not in the document]
Comment (jlucaso1) [reply tools/oracle-core/tests/incoming_orientation_probe.rs:14]: I found and corrected two stale sentences in agent_docs/voip_media_oracle.md in faee7f54. They contradicted the scoped instructions at tools/oracle-core/AGENTS.md:19-21, which allow an absent implicit cache to skip and require explicit overrides or corrupt captures to fail. The probe now follows that same documented policy; reverting to expect would reintroduce the reproduced three-test CI failure.

No assertion was removed. I reran the incoming and orientation probes with WA_WASM_DIR pointing to the pinned real capture: all five tests executed without skips, including 3,840 wire packets, 285 element-boundary cases, eight passthrough pairs, 256 receive-renderer metadata values and ten outbound cases. A run that prints skipping remains explicitly excluded from conformance evidence.

File: tools/xtask/src/size_baseline.rs [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+            run.id
+        );
+        let pages: Vec<Artifacts> =
+            serde_json::from_slice(&gh(&["api", "--paginate", "--slurp", &endpoint])?)?;
+        let artifacts: Vec<_> = pages
+            .iter()
+            .flat_map(|page| &page.artifacts)
+            .filter(|artifact| artifact.name == "size-metrics")
+            .collect();
+        let [artifact] = artifacts.as_slice() else {
```
[the remark this answers is not in the document]
Comment (jlucaso1) [reply tools/xtask/src/size_baseline.rs:160]: Fixed in 876fa85b. Artifact provenance and current upload-interval filtering now happen before uniqueness checking. The selected immutable artifact ID is passed to the download path, which uses the artifact-ID endpoint and extracts only the two fixed JSON members.

Added a regression with old and current same-name artifacts in both listing orders. It failed before the fix and now selects only the current ID. Same-attempt ambiguity, invalid provenance, expiration and compiler mismatch still fail closed. All 18 tooling tests pass. I also executed the new download path against the real base artifact for 47e1b5b41 and validated the downloaded metadata.
